presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitPartitionInfo.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hive;
+
+import org.openjdk.jol.info.ClassLayout;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+
+import static io.airlift.slice.SizeOf.sizeOfObjectArray;
+
+public class HiveSplitPartitionInfo
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(HiveSplitPartitionInfo.class).instanceSize();
+    private static final int INTEGER_INSTANCE_SIZE = ClassLayout.parseClass(Integer.class).instanceSize();
+
+    private final Properties schema;
+    private final List<HivePartitionKey> partitionKeys;
+    private final String partitionName;
+    private final Map<Integer, HiveTypeName> columnCoercions;
+    private final Optional<HiveSplit.BucketConversion> bucketConversion;
+
+    public HiveSplitPartitionInfo(
+            Properties schema,
+            List<HivePartitionKey> partitionKeys,
+            String partitionName,
+            Map<Integer, HiveTypeName> columnCoercions,
+            Optional<HiveSplit.BucketConversion> bucketConversion)
+    {
+        this.schema = schema;
```

Comment:
nit: do we want `requireNonNull` for them?

Reply:
This does not seem addressed

Reply:
yup i moved the requireNonNulls to the top of the constructor because path would have been ugly since it calls a method.